src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsReconciliationService.kt

```diff
@@ -34,6 +34,25 @@ class AdjudicationsReconciliationService(
     val log: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
+  suspend fun generateAdjudicationsReconciliationReport() {
+    val activePrisonersCount = nomisApiService.getActivePrisoners(0, 1).totalElements
+
+    telemetryClient.trackEvent("adjudication-reports-reconciliation-requested", mapOf("active-prisoners" to activePrisonersCount.toString()))
+    log.info("Adjudication reconciliation report requested for $activePrisonersCount active prisoners")
+
+    runCatching { generateReconciliationReport(activePrisonersCount) }
+      .onSuccess {
+        log.info("Adjudication reconciliation report completed with ${it.size} mismatches")
+        telemetryClient.trackEvent("adjudication-reports-reconciliation-report", mapOf("mismatch-count" to it.size.toString(), "success" to "true") + it.asMap())
+      }
+      .onFailure {
+        telemetryClient.trackEvent("adjudication-reports-reconciliation-report", mapOf("success" to "false"))
+        log.error("Adjudication reconciliation report failed", it)
+      }
+  }
+
+  private fun List<MismatchAdjudicationAdaPunishments>.asMap(): Map<String, String> = this.associate { it.prisonerId.offenderNo to ("${it.nomisAda.count}:${it.dpsAdas.count},${it.nomisAda.days}:${it.dpsAdas.days}") }
```

Comment:
The pattern seems to be copying the entire method from the existing Resource, but removing the special CoroutineScope `reportScope` (which was used to "fire and forget" but we don't want to forget about them any more).